sshd-common/src/main/java/org/apache/sshd/common/util/io/IoUtils.java

```diff
@@ -426,29 +426,40 @@ public static Boolean checkFileExistsAnySymlinks(Path path, boolean neverFollowS
                 // this is a bad fix because this leaves a nasty race condition - the directory may turn into a symlink
                 // between this check and the call to open()
                 for (int i = 1; i <= path.getNameCount(); i++) {
-                    Path checkForSymLink = getPartialPath(path, i);
+                    Path checkForSymLink = getFirstPartsOfPath(path, i);
                     BasicFileAttributes basicFileAttributes
                             = Files.readAttributes(checkForSymLink, BasicFileAttributes.class, LinkOption.NOFOLLOW_LINKS);
                     if (basicFileAttributes.isSymbolicLink()) {
                         return false;
                     }
                 }
             }
-            return Boolean.TRUE;
+            return true;
         } catch (NoSuchFileException e) {
-            return Boolean.FALSE;
+            return false;
```

Comment:
I have a problem with the `Boolean` return type. Using a Boolean for a tri-state return is confusing. Also, the javadoc would need to spell out what a `null` return means. Can't this just return `boolean`,  and just return false on IOE? As far as I see, callers map this to an AccessDeniedException anyway.